src/reconcile/rec_write.c

```diff
@@ -1113,12 +1119,13 @@ __rec_child_modify(WT_SESSION_IMPL *session,
 			 * process will have completed before we walk any pages
 			 * for checkpoint.
 			 */
-			if ((ret = __wt_page_in(session, ref,
-			    WT_READ_CACHE | WT_READ_NO_EVICT |
-			    WT_READ_NO_GEN | WT_READ_NO_WAIT)) == WT_NOTFOUND) {
+			ret = __wt_page_in(session, ref,
+			    WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN);
+			if (ret == WT_NOTFOUND) {
```

Comment:
I think this may in fact be the real bug.  With `WT_READ_NO_WAIT`, any child page that happens to be flipped to `WT_REF_LOCKED` (even if only temporarily before eviction decides not to proceed) would return `WT_NOTFOUND` here and be treated as clean when reconciling the parent.